sql/fxa_derived/fxa_amplitude_export_v1/query.sql

```diff
@@ -0,0 +1,100 @@
+WITH
+  base_events AS(
+  SELECT
+    *
+  FROM
+    `moz-fx-fxa-prod-0712.fxa_prod_logs.docker_fxa_auth_20*`
+  WHERE
+    _TABLE_SUFFIX = FORMAT_DATE('%g%m%d', @submission_date)
+    AND jsonPayload.fields.event_type IN ( 'fxa_activity - cert_signed',
+      'fxa_activity - access_token_checked',
+      'fxa_activity - access_token_created' )
+    AND jsonPayload.fields.user_id IS NOT NULL ),
+  grouped_by_user AS (
+  SELECT
+    -- to prevent weirdness from timestamp field, use provided
+    -- submission date parameter as timestamp
+    CAST(@submission_date AS DATETIME) AS submission_timestamp,
+    -- TODO: JS udf with HMAC
```

Comment:
I had been assuming we'd need to create a custom python script to do the HMACing outside of BQ because there's no clear path to making the key available inside BQ.

Do you have a vision in mind for how this JS UDF would work and how we make the key available without spilling the key contents into query logs, etc?

Reply:
It may be possible to make a small JS lib containing the HMAC key and store that in a GCS blob that it only accessible by the Airflow service account. We could then define a JS UDF that declares that lib in GCS as a dependency. I think that would avoid the actual content of the key ever appearing in logs, and it would allow only the Airflow user access to the secret.

I'm also not confident, though, that this will be able to scale. We have encountered various resource exceeded errors when attempting to use a JS function over a large number of rows.

Reply:
I imagine it's simpler to create a locked-down dataset and edit [access control to those query logs](https://cloud.google.com/logging/docs/access-control). Some details of this may need to be worked out, but that would alleviate having to create big work-arounds for this key, and allow us to use secrets more easily in the future as well.

Reply:
Opened up https://bugzilla.mozilla.org/show_bug.cgi?id=1634179 to discuss this with ops.